modules/covid_vaccine/app/models/covid_vaccine/v0/expanded_registration_submission.rb

```diff
@@ -19,6 +50,60 @@ class ExpandedRegistrationSubmission < ApplicationRecord
       attr_encrypted :raw_form_data, key: Settings.db_encryption_key, marshal: true, marshaler: JsonMarshal::Marshaller
       attr_encrypted :eligibility_info, key: Settings.db_encryption_key, marshal: true,
                                         marshaler: JsonMarshal::Marshaller
+
+      # We want records that have acceptable discharge status, country, state, zip and facility
+      def self.to_csv(separator = '^')
+        CSV.generate(col_sep: separator) do |csv|
+          eligible_us.order('created_at DESC').each do |es|
+            csv << es.send(:csv_row).map { |e| e&.delete('"^') }
+          end
+        end
+      end
+
+      private
+
+      def country_us?
+        raw_form_data[:country_name] == 'USA'
+      end
+
+      def csv_row
+        raw_form_data
+          .values_at('first_name', 'middle_name', 'last_name')
+          .append(csv_birth_date)
+          .append(raw_form_data['ssn'])
+          .append(csv_birth_sex)
+          .append(icn)
+          .append(csv_address)
+          .concat(raw_form_data.values_at('city', 'state_code', 'zip_code'))
+          .append(csv_phone)
+          .append(raw_form_data['email_address'])
+          .append(csv_preferred_facility)
+          .append(VA_AGENCY_IDENTIFIER)
+      end
+
+      def csv_preferred_facility
+        raw_form_data['preferred_facility'].delete_prefix('vha_')
```

Comment:
There will be nil preferred_facilities - for international users or those who for some reason enter an invalid zip. So this should be robust to nil entry and return an empty string.

Reply:
handled with &. in new refactor